portal-api/src/v1/users/sign-in-link.service.js

```diff
@@ -17,37 +18,51 @@ class SignInLinkService {
   async createAndEmailSignInLink(user) {
     const { _id: userId, email: userEmail, firstname: userFirstName, surname: userLastName } = user;
 
-    const signInCode = this.#createSignInCode();
-    await this.#saveSignInCodeHashAndSalt({ userId, signInCode });
+    const signInToken = this.#createSignInToken();
+    await this.#saveSignInTokenHashAndSalt({ userId, signInToken });
 
     return this.#sendSignInLinkEmail({
-      signInLink: `${PORTAL_UI_URL}/login/sign-in-link?t=${signInCode}`,
+      signInLink: `${PORTAL_UI_URL}/login/sign-in-link?t=${signInToken}`,
       userEmail,
       userFirstName,
       userLastName,
     });
   }
 
-  #createSignInCode() {
+  async isValidSignInToken({ username, signInToken }) {
+    return new Promise((resolve, reject) => {
+      findByUsername(username, (e, user) => {
+        if (e || !user) {
+          const error = new Error(`Failed to find user: ${username}`);
+          error.cause = e || 'User object not defined';
+          reject(error);
+        }
+        const { hash, salt } = user.signInToken;
```

Comment:
We'll reject with an unexpected error here if `user.signInToken` is `undefined` - is that what we want?